### PR TITLE
Feat/new payment be

### PR DESCRIPTION
**Pull request by muhammed**

### CODE REVIEW

File: lerna-debug.log [hunk 1]
```diff
@@ -0,0 +1,7 @@
+54 error Error: Command failed with exit code 128: git tag @heliofi/solana-adapter@3.0.7-alpha -m @heliofi/solana-adapter@3.0.7-alpha
+54 error fatal: tag '@heliofi/solana-adapter@3.0.7-alpha' already exists
+54 error     at makeError (/Users/muhammederdem/works/helio/heliopay/node_modules/execa/lib/error.js:60:11)
+54 error     at handlePromise (/Users/muhammederdem/works/helio/heliopay/node_modules/execa/index.js:118:26)
+54 error     at processTicksAndRejections (node:internal/process/task_queues:96:5)
+54 error     at async Promise.all (index 1)
+54 error     at async Promise.all (index 0)
```
Comment (00MB) [inline lerna-debug.log:7]: Should not stage lerna-debug.log, put it in .gitignore

File: packages/react/src/components/heliopay-container/index.tsx [hunk 2]
```diff
@@ -124,7 +165,66 @@ const HelioPayContainer: FC<HeliopayContainerProps> = ({
...
+      getCountry();
+    }
+
+    if (
+      paymentDetails?.currency != null &&
+      paymentDetails?.normalizedPrice != null
+    ) {
+      if (
+        paymentDetails?.fixedCurrency &&
+        paymentDetails?.features?.requireFixedCurrency
```
Comment (00MB) [inline packages/react/src/components/heliopay-container/index.tsx:182]: Do we even have requireFixedCurrency on dynamic links features object? I think you should use the prop passed through

File: packages/react/src/components/heliopay-container/index.tsx [hunk 3]
```diff
@@ -79,26 +91,55 @@ const HelioPayContainer: FC<HeliopayContainerProps> = ({
...
+          amount: paymentDetails?.fixedCurrency?.price,
+        });
+      }, expirationMS - nowMS);
+    }
+  }, [
+    tokenExpiration,
+    paymentDetails?.fixedCurrency,
+    paymentDetails?.currency,
+    paymentDetails?.fixedCurrency,
+    expirationMS,
```
Comment (00MB) [inline packages/react/src/components/heliopay-container/index.tsx:140]: Maybe am wrong here but just make sure paymentDetails is coming from HelioPay props not paymentRequest details

File: packages/react/src/domain/services/JWTService.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export class JWTService {
+  static decodeToken(token: string): {
+    rate: number;
+    exp: number;
+    iat: number;
+  } {
+    return JSON.parse(Buffer.from(token?.split('.')[1], 'base64').toString());
+  }
+}
```
Comment (00MB) [inline packages/react/src/domain/services/JWTService.ts:9]: I think we removed this and used a jwt decode library with mapping to types. Might be best to use that instead as not all decode tokens are going to have this type. For example swap token will not have a rate

File: packages/react/src/infrastructure/solana-utils/SignTransaction.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      );
+      signedTransaction.addSignature(
+        otherSigner.publicKey,
+        Buffer.from(otherSignature)
+      );
+    }
+    return JSON.stringify(signedTransaction.serialize());
+  } catch (error) {
+    return '';
+  }
```
Comment (00MB) [inline packages/react/src/infrastructure/solana-utils/SignTransaction.ts:28]: Why don't you return the error?

**Review by 00MB — CHANGES_REQUESTED** [review]

File: packages/react/src/infrastructure/solana-utils/payment/paylink/PaylinkSubmitService.ts [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+  ApproveTransactionResponse
+> {
+  protected readonly endpoint = 'transaction/submit';
+
+  protected readonly prepareEndpoint = '/prepare/transaction/sol/paylink';
+
+  protected async init(props: CreatePaymentProps): Promise<void> {
+    await super.init(props);
+  }
+  // paymentRequestId: string,
```
Comment (martinavagyan) [inline packages/react/src/infrastructure/solana-utils/payment/paylink/PaylinkSubmitService.ts:78]: commented code

File: packages/react/src/infrastructure/solana-utils/payment/paylink/PaylinkSubmitService.ts [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+          body: JSON.stringify({
+            paymentRequestId,
+            currency: symbol,
+            quantity: quantity,
+            fixedCurrencyRateToken: rateToken,
+            ...request,
+          }),
+        },
+      });
+    // const prepareTransactionResponse =
```
Comment (martinavagyan) [inline packages/react/src/infrastructure/solana-utils/payment/paylink/PaylinkSubmitService.ts:210]: remove

File: packages/react/src/components/customer-details-form-modal/index.tsx [hunk 8]
```diff
@@ -240,16 +248,58 @@ const CustomerDetailsFormModal = ({
...
-                          values.quantity
-                        )}{' '}
-                        {activeCurrency?.symbol}
-                      </b>
+                      <div>
+                        Total price:{' '}
+                        <b>
+                          {activeCurrency &&
+                            CurrencyService.getCurrencyBySymbol(
+                              activeCurrency?.symbol as string
```
Comment (martinavagyan) [inline packages/react/src/components/customer-details-form-modal/index.tsx:260]: String(activeCurrency?.symbol)

File: packages/react/src/components/customer-details-form-modal/index.tsx [hunk 9]
```diff
@@ -240,16 +248,58 @@ const CustomerDetailsFormModal = ({
...
+                            values.quantity
+                          )}{' '}
+                          {activeCurrency?.symbol}
+                          {fixedPrice && requireFixedCurrency && fixedCurrency && (
+                            <span>
+                              {' '}
+                              (
+                              {
+                                CurrencyService.getCurrencyBySymbol(
+                                  fixedCurrency as string
```
Comment (martinavagyan) [inline packages/react/src/components/customer-details-form-modal/index.tsx:273]: String(fixedCurrency)

File: packages/react/src/components/customer-details-form-modal/index.tsx [hunk 10]
```diff
@@ -240,16 +248,58 @@ const CustomerDetailsFormModal = ({
...
+                                )?.symbolPrefix
+                              }
+                              {formatTotalPrice(
+                                TokenConversionService.convertFromMinimalUnits(
+                                  CurrencyService.getCurrencyBySymbol(
+                                    fixedCurrency as string
```
Comment (martinavagyan) [inline packages/react/src/components/customer-details-form-modal/index.tsx:279]: String(fixedCurrency)

File: packages/react/src/infrastructure/helio-api/HelioApiAdapter.ts [hunk 11]
```diff
@@ -123,4 +129,59 @@ export const HelioApiAdapter = {
...
+    cluster: Cluster;
+    options: FetchOptions;
+    shouldParseJson?: boolean;
+  }): Promise<T> {
+    const HELIO_BASE_API_URL = getHelioApiBaseUrl(cluster);
+    const url = `${HELIO_BASE_API_URL}${endpoint}`;
+    const contentType = !options.clearContentType
+      ? { 'Content-Type': 'application/json' }
+      : null;
+    const response = await fetch(url, {
```
Comment (martinavagyan) [inline packages/react/src/infrastructure/helio-api/HelioApiAdapter.ts:175]: why don't you use our fetch middleweare ?

File: packages/react/src/infrastructure/helio-api/HelioApiAdapter.ts [hunk 12]
```diff
@@ -1,10 +1,16 @@
+import { Currency } from '@heliofi/common';
```
Comment (martinavagyan) [inline packages/react/src/infrastructure/helio-api/HelioApiAdapter.ts:1]: you can just reuse HelioApiAdapter here

File: packages/react/src/domain/services/JWTService.ts (hunk 4 above)
Comment (martinavagyan) [reply packages/react/src/domain/services/JWTService.ts:9]: agree we should just use the same lib we use in dash, and BE

**Review by martinavagyan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**